core/app/models/spree/app_configuration.rb

```diff
@@ -365,6 +365,17 @@ def tax_adjuster_class
       @tax_adjuster_class ||= Spree::Tax::OrderAdjuster
     end
 
+    # Allows providing your own class for generating order numbers.
+    #
+    # @!attribute [rw] order_number_generator
+    # @return [Class] a class with the same public interfaces as
+    #   Spree::Order::NumberGenerator
+    # @api experimental
+    attr_writer :order_number_generator
+    def order_number_generator
+      @order_number_generator ||= Spree::Order::NumberGenerator
```

Comment:
One thought: maybe this should be an instance of `NumberGenerator` rather than the class. This would allow developers to specify a length, prefix, or letters without needing to write their own class